src/tuned_models.jl

```diff
@@ -278,27 +291,134 @@ function event(metamodel,
                          verbosity,
                          tuning,
                          history,
                          state,
                          acceleration::CPU1)
-    map(metamodels) do m
-        event(m, resampling_machine, verbosity, tuning, history, state)
-    end
+     local ret
+     resampling_machine = resampling_machines[1]
+     n_metamodels = length(metamodels)
+     verbosity < 1 || begin
+                 p = Progress(n_metamodels,
+                 dt = 0,
+                 desc = "Evaluating over $(n_metamodels) metamodels: ",
+                 barglyphs = BarGlyphs("[=> ]"),
+                 barlen = 25,
+                 color = :yellow)
+                 update!(p,0)
+      end
+    
+      @sync begin   
+        ret = map(metamodels) do m
+            r= event(m, resampling_machine, verbosity, tuning, history, state)
+            verbosity < 1 || next!(p)
+            r
+       end
+      end
+
+    return ret
 end
 
 function assemble_events(metamodels,
-                         resampling_machine,
+                         resampling_machines,
                          verbosity,
                          tuning,
                          history,
                          state,
                          acceleration::CPUProcesses)
-    pmap(metamodels) do m
-        event(m, resampling_machine, verbosity, tuning, history, state)
+  resampling_machine = resampling_machines[1]
+
+  ret = if verbosity < 1
+       pmap(metamodels) do m
+            event(m, resampling_machine, verbosity, tuning, history, state)
+       end
+  else
+      n_metamodels = length(metamodels)
+      p = Progress(n_metamodels,
+                 dt = 0,
+                 desc = "Evaluating over $(n_metamodels) metamodels: ",
+                 barglyphs = BarGlyphs("[=> ]"),
+                 barlen = 25,
+                 color = :yellow)
+      update!(p,0)
+      progress_pmap(metamodels, progress=p) do m
+            event(m, resampling_machine, verbosity, tuning, history, state)
+      end
+
+  end
+    
+    return ret
+end
+
+@static if VERSION >= v"1.3.0-DEV.573"
+# one machine for each thread; cycle through available threads:
+function assemble_events(metamodels,
+                         resampling_machines,
+                         verbosity,
+                         tuning,
+                         history,
+                         state,
+                         acceleration::CPUThreads)
+    
+    if Threads.nthreads() == 1
+        return assemble_events(metamodels,
+                         resampling_machines,
+                         verbosity,
+                         tuning,
+                         history,
+                         state,
+                         CPU1())
+   end
+    n_metamodels = length(metamodels)
+    n_threads = Threads.nthreads()
+    
+    
+    ret = Array{Any, 1}(undef, n_metamodels)
+    verbosity < 1 || (p = Progress(n_metamodels,
```

Comment:
Be nice if we could avoid this abstract `Any` typing of `ret`, but I'm sure you've thought about that. I doubt it's a real performance issue, except possibly when dealing with very small data and huge numbers of models (hyper parameters) being evaluated.

Reply:
`ret = Vector{Tuple{M,Any}(undef, n_metamodels)` is the best i can do since the type of `result` isn't constrained.

Reply:
No that's fine.